lib/cocoapods/installer/project_cache/target_cache_key.rb

```diff
@@ -0,0 +1,134 @@
+module Pod
+  class Installer
+    module ProjectCache
+      # Uniquely identifies a Target.
+      #
+      class TargetCacheKey
+        require 'cocoapods/target/pod_target.rb'
+        require 'cocoapods/target/aggregate_target.rb'
+        require 'digest'
+
+        # @return [Symbol]
+        #         The type of target. Either aggregate or pod target.
+        #
+        attr_reader :type
+
+        # @return [Hash{String => Object}]
+        #         The hash containing key-value pairs that identify the target.
+        #
+        attr_reader :hash
```

Comment:
should rename, since `Object#hash` returns the hash value when used as a dictionary key